File: python/baekjoon/dfs&bfs/1260.py
import sys
from collections import deque
n, m, start= map(int, sys.stdin.readline().split())

nodes = []
for _ in range(m):
    nodes.append(list(map(int,sys.stdin.readline().split())))

graph=[[] for _ in range(n+1)]
for node in nodes:
    graph[node[0]].append(node[1])
    graph[node[1]].append(node[0])

graph=list(map(sorted,graph))

def dfs(graph, v, visited):
    visited[v] = True
    print(v, end=' ')
    for i in graph[v]:
        if not visited[i]:
            dfs(graph, i, visited)
    

def bfs(graph, start, visited):
    queue = deque([start])
    visited[start] = True
    while queue:
        v = queue.popleft()
        print(v, end = ' ')
        for i in graph[v]:
            if not visited[i]:
                queue.append(i)
                visited[i]=True



visited = [False]*(n+1)
visited2 = [False]*(n+1)

dfs(graph, start, visited)
print()
bfs(graph, start, visited2)
print()

File: cpp/dfs_bfs/1260.cpp
#include <iostream>
#include <queue>

using namespace std;

int n, m, v;
bool check[1001] = {false};
int graph [1001][1001];
queue<int> q;

void dfs(int v){
    cout << v << ' ';
    for(int i=1;i<=n;i++){
        if(!check[i] && graph[v][i] == 1){
            check[i] = 1;
            dfs(i);
        }
    }
}

void bfs(int v){
    q.push(v);
    check[v] = 1;
    while(!q.empty()){
        v = q.front();
        q.pop();
        cout << v << ' ';
        for(int i=1;i<=n;i++){
            if(!check[i] && graph[v][i] == 1){
                check[i] = 1;
                q.push(i);
            }
        }
    }
}


int main(){
    cin >> n >> m >> v;
    for(int i=0;i<m;i++){
        int u,v;
        cin >> u >> v;
        graph[u][v] = graph[v][u] = 1;
    }
    check[v] = 1;
    dfs(v);
    cout << '\n';
    for(int i=1;i<1001;i++){
        check[i] = false;
    }
    bfs(v);
    cout << '\n';
}